### PR TITLE
Schema update

### PR DESCRIPTION
**Pull request by kiverso**
Simplifies the database schema and updated the podcast application process to create a new podcast and podcast_tags after the application table was removed. Updated specs for the affected classes.

### CODE REVIEW

File: app/controllers/podcasts_controller.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ def create
     user = User.find(params[:user_id])
     
     podcast = user.podcasts.new(podcast_params)
-    binding.pry
     if podcast.save
+      params[:podcast][:tags].each {|tag| PodcastTag.create({podcast_id: podcast.id, tag_id: tag})}
```
Comment (muydanny) [inline app/controllers/podcasts_controller.rb:12]: tag passes in an array of tag ids

File: app/models/following.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 class Following < ApplicationRecord
-  belongs_to :podcast
+  belongs_to :favorite_podcast, foreign_key: 'podcast_id', class_name: 'Podcast'
```
Comment (muydanny) [inline app/models/following.rb:2]: Might need to be tested more vigorously. not currently breaking but created an alias

File: app/views/podcasts/new.html.erb [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+<%= form_for ([:user, @podcast]) do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name %> <br>
 
   <%= f.label :location %>
   <%= f.select :location, ['Denver', 'NYC', 'Dallas', 'San Francisco'], :include_blank => true %> <br>
 
   <%= f.label :tags %>
-  <%= f.collection_check_boxes(:tags, Tag.all, :id, :name) %> <br>
+  <%= f.collection_check_boxes(:tags, Tag.all, :id, :name, include_hidden: false) %> <br>
```
Comment (kiverso) [inline app/views/podcasts/new.html.erb:9]: include_hidden: false keeps us from passing in unchecked checkboxes.

File: db/schema.rb [hunk 4]
```diff
@@ -10,30 +10,11 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_07_23_191204) do
+ActiveRecord::Schema.define(version: 2020_07_22_010904) do
 
   # These are extensions that must be enabled in order to support this database
```
Comment (muydanny) [inline db/schema.rb:15]: Rebuilt the schema. Deleted everything we had discussed.

File: db/migrate/20200722004638_create_users.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@ def change
       t.string :user_name
       t.string :access_token
       t.string :refresh_token
-      t.integer :role, default: 0
+      t.boolean :admin, default: false
```
Comment (janegreene) [inline db/migrate/20200722004638_create_users.rb:7]: Very clean and easy.

**Pull request merged**